components/screens/BlogScreen/BlogScreen.js

```diff
@@ -1,62 +1,98 @@
+    else Linking.openURL(urlToOpen)
+  } catch (error) {
+    console.log(error.message)
+  }
+}
+
 const BlogScreen = () => {
 
   const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
   const isFocused = useIsFocused();
 
   const getData = async () => {
+    setLoading(true);
     const resp = await fetch("https://cascade-api.herokuapp.com/blog");
     const data = await resp.json();
     setData(data);
+    setLoading(false);
   };
   useEffect(() => {
-    if (isFocused) {
-      getData();
+    getData();
 
-    }
-
-  }, [isFocused]);
+  }, []);
 
   return (
-    <SafeAreaView>
+    <SafeAreaView style={Styles.container}>
       <ScrollView>
+        {
+          loading ?
+            (
+              <View style={Styles.container}>
+                <ActivityIndicator
+                  animating={true}
+                  visible={loading}
+                  size="large"
+                  color="#0176ae"
+                  textContent={'Loading...'} />
+              </View>
+
+            )
 
-        {data.map(({ title, postDate, body, url }) => (
-          <View key={url}>
-            <Text key={title} style={Styles.title}>
-              {title}
-            </Text>
-            <Text style={Styles.date}
-              key={postDate}>{postDate}</Text>
-            <Image key={'rc34'}
-              source={{ uri: 'https://cbc-drupal-assets.s3.amazonaws.com/styles/blog_image/s3/blog/leafy%20and%20dank%20%283%29.JPG?R2Pc9TRNWsiPfO0asl_VSPULtFDI4Dqs&itok=O8FLO9W1' }}
-              style={Styles.image}
+            :
+            data.map(({ title, postDate, body, url, image }) => (
+              <View key={url}>
+                <Text key={title} style={Styles.title}>
+                  {title}
+                </Text>
+                <Text style={Styles.date}
+                  key={postDate}>{postDate}</Text>
+                <Image key={image}
+                  source={{ uri: image }}
+                  style={Styles.image}
 
-            />
-            <Text key={body} style={Styles.blogItem} >{body}</Text>
-            <View key={url} style={{ width: 200, height: 50 }}>
-              <Button
-                style={Styles.button}
-                type="solid"
-                title="Read more"
-                color="#F55243"
-                size={6}
-                onPress={(e) => {
-                  e.preventDefault();
-                  InAppBrowser.open('http://www.cascade.org' + url);
-                }}
-              />
-            </View>
-          </View>
-        ))
+                />
+                <Text key={body} style={Styles.blogItem} >{body}</Text>
+                <View key={url} style={{ width: 200, height: 50 }}>
+                  <Button
+                    style={Styles.button}
+                    type="solid"
+                    title="Read more"
+                    color="#F55243"
+                    size={6}
+                    onPress={(e) => {
```

Comment:
As a suggestion, type -> color could be moved to the Styles sheet, as button or button2.

Reply:
Thank you Natalia, moved it to the styles sheet!